backend_modules/azure/network/main.tf

```
@@ -0,0 +1,203 @@
+  priority                    = 110
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = "443"
+  source_address_prefix       = "*"
+  destination_address_prefix  = "*"
+}
+
+resource "azurerm_subnet_network_security_group_association" "public-nsg-association" {
+  count = var.create_network? 1: 0
+
+  subnet_id                 = "${azurerm_subnet.public-sn[0].id}"
+  network_security_group_id = "${azurerm_network_security_group.public-nsg[0].id}"
+}
+
+ resource "azurerm_network_security_group" "private-nsg" {
+  count = var.create_network? 1: 0
+
+  name                = "${var.name_prefix}-private-nsg"
+  resource_group_name = "${azurerm_resource_group.suma-rg.name}"
+  location            = "${azurerm_resource_group.suma-rg.location}"
+} 
+
+resource "azurerm_network_security_group" "private-additional-nsg" {
+  count = var.create_network? 1: 0
+
+  name                = "${var.name_prefix}-private-nsg"
+  resource_group_name = "${azurerm_resource_group.suma-rg.name}"
+  location            = "${azurerm_resource_group.suma-rg.location}"
+}
+
+resource "azurerm_subnet_network_security_group_association" "private-additonal-nsg-association" {
+  count = var.create_network? 1: 0
+
+  subnet_id                 = "${azurerm_subnet.private-additional-sn[0].id}"
+  network_security_group_id = "${azurerm_network_security_group.private-additional-nsg[0].id}"
+}
+
+resource "azurerm_subnet_network_security_group_association" "private-nsg-association" {
+  count = var.create_network? 1: 0
+
+  subnet_id                 = "${azurerm_subnet.private-sn[0].id}"
+  network_security_group_id = "${azurerm_network_security_group.private-nsg[0].id}"
+}
+
+resource "azurerm_public_ip" "nat-pubIP" {
+  count = var.create_network? 1: 0
+
+  name                = "nat-gateway-pubIP"
+  resource_group_name = "${azurerm_resource_group.suma-rg.name}"
+  location            = "${azurerm_resource_group.suma-rg.location}"
+  allocation_method   = "Static"
+  sku                 = "Standard"
+  zones               = ["1"]
+}
+
+resource "azurerm_nat_gateway" "suma-ngw" {
+  count = var.create_network? 1: 0
+
+  name                = "${var.name_prefix}-ngw"
+  resource_group_name = "${azurerm_resource_group.suma-rg.name}"
+  location            = "${azurerm_resource_group.suma-rg.location}"
+  public_ip_address_ids   = ["${azurerm_public_ip.nat-pubIP[0].id}"]
+}
+
+resource "azurerm_subnet_nat_gateway_association" "suma-ngw-association" {
+  count = var.create_network? 1: 0
+
+  subnet_id      = "${azurerm_subnet.private-sn[0].id}"
+  nat_gateway_id = "${azurerm_nat_gateway.suma-ngw[0].id}"
+}
+
+output "configuration" {
+  depends_on = [azurerm_subnet_route_table_association.private-rtas, azurerm_subnet_route_table_association.private_additional-rtas , azurerm_subnet_route_table_association.public-rtas]
```

the resource name is different from the one defined above. https://github.com/uyuni-project/sumaform/pull/814/files#diff-cb7f71c29f6b3f79b7e7e22c7796c9960887631cbedec34d98b96405e2c74827R80
```suggestion
  depends_on = [azurerm_subnet_route_table_association.private-rtas, azurerm_subnet_route_table_association.private-additional-rtas , azurerm_subnet_route_table_association.public-rtas]
```